src/realizations/catchment/Realization_Config.cpp

```diff
@@ -0,0 +1,291 @@
+#include "Realization_Config.hpp"
+
+#include "Tshirt_Realization.hpp"
+#include <Simple_Lumped_Model_Realization.hpp>
+#include "tshirt_params.h"
+#include <iostream>
+#include <regex>
+#include <dirent.h>
+
+using namespace realization;
+
+std::shared_ptr<Simple_Lumped_Model_Realization> Realization_Config_Base::get_simple_lumped() {
```

Comment:
Going to have to think of a way to scale this as we get new realizations we want to support.

Reply:
This is something I was looking into this morning. I wouldn't say it doesn't scale as much as it isn't sustainable.

My first goal is to get rid of the get_XXX functions for the realizations. They are only there to simplify the `get_realization()` function. The solution is probably just to make different constructors for each realization. That would also allow us to move some more content out of `Reader_Config.hpp`

Reply:
I was thinking we could push a constructor to the realization that takes the `options` and does the unpack there.  Then we could simplify to an enum of supported realizations and a switch to construct the one the config is asking for?